src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1547,7 +1558,9 @@ void TorrentHandle::handleTrackerWarningAlert(const lt::tracker_warning_alert *p
     const QString message = p->warning_message();
 
     // Connection was successful now but there is a warning message
-    m_trackerInfos[trackerUrl].lastMessage = message; // Store warning message
+    TrackerEntry entry;
+    if (tracker(trackerUrl, entry))
```

Comment:
@arvidn
Maybe `tracker_alert` could include an `announce_entry` field (or just a pointer to it) in addition to `tracker_url()`? So that we can use it directly instead of going back and search it in `torrent_handle::trackers()`. Does it make sense?
https://www.libtorrent.org/reference-Alerts.html#tracker_alert

Reply:
it's not clear to me what information you need out of the `announce_entry`. The tracker alert *could* contain an announce entry field (not a pointer though, since it can't spill its guts by pointing to internal objects). But it seems the URL is the most relevant part of it. And technically, it would have to be an `announce_endpoint`, right?

Reply:
> And technically, it would have to be an announce_endpoint, right?

Yes, it would have to be `announce_endpoint`, because those are reporting errors. And we would still need to search the related `announce_entry` and check its status, because we want to hide endpoint errors if at least one is working. So unless you report error only if all endpoint failed, there would be no benefit of proposed change.